.github/workflows/releases.yml

```diff
@@ -0,0 +1,56 @@
+name: Release
+
+on:
+  workflow_run:
+    workflows: ["tests-types"]
+    branches: 
+      push: 
+        - main
+    types:
+      - completed
+  
+jobs:
+  release:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+
+      - name: Setup Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+
+      - name: Install Dependencies
+        run: npm ci --prefer-offline
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          # This expects you to have a script called release which does a build for your packages and calls changeset publish
+          publish: npm changeset:publish
```

Comment:
I would much rather this be an explicit call to changesets, with a `--tag beta`, instead of calling an npm script.

Reply:
Sounds good